morepath/dispatch.py

```diff
@@ -0,0 +1,73 @@
+import reg
+from functools import wraps, partial
+from .reify import reify
+
+
+class delegate(object):
+
+    def __init__(self, *predicates):
+        self.make_generic = reg.dispatch(*predicates)
+        self.external_predicates = False
+
+    def __call__(self, func):
+
+        def install(reg):
+            setattr(reg, func.__name__, self.make_generic(func))
+
+        @wraps(func)
+        @reify
+        def delegator(app):
+            actual = getattr(app.config.reg_registry, func.__name__)
+            result = partial(actual, app)
+            result.component_key_dict = actual.component_key_dict
+            return result
+
+        delegator.external_predicates = self.external_predicates
+        delegator.install_delegate = install
+        return delegator
+
+    @classmethod
+    def on_external_predicates(cls):
+        instance = cls()
+        instance.make_generic = reg.dispatch_external_predicates()
+        instance.external_predicates = True
+        return instance
+
+
+class RegRegistry(object):
+    """A registry to group together the implementation of delegated
+    methods of one application.
+
+    """
+
+    factory_arguments = {'installers': list}
```

Comment:
Okay, I think I get the special config object now. On it you generate an installers list from the information on ``app_class``, which is then a configuration object which then gets inserted here. That's far too magical for my tastes. But I think ``app_class_arg`` can make it all go away by doing that inspection here instead, which would be fine.

Interestingly I do need to do such inspection myself but for cleanup, not for registration, as on my branch the registries are directly stored on the app class as opposed of inside ``config``.